lib/composites/menu/events/main.js

```diff
@@ -14,25 +14,29 @@ const ACTIVE_CLASS = 'dqpl-active';
 
 module.exports = (elements) => {
   /**
-   * Listen for clicks outside the menu (when
-   * its opened AND not locked) to close it
+   * Handle for clicks outside sideBar and dropdowns within topBar and close them if needed
    */
-  if (elements.menu && elements.trigger) {
+
+  if (elements.menu && elements.trigger && elements.topBar) {
     document.addEventListener('click', (e) => {
       const target = e.target;
       const isLocked = elements.menu.getAttribute('data-locked') === 'true';
-      const isWithin = closest(target, '.dqpl-side-bar');
+      const isWithinSideBar = closest(target, '.dqpl-side-bar');
+      const isWithinDropdown = closest(target, '.dqpl-top-bar li[data-dropdown-expanded]');
       const isHamburger = Classlist(target).contains('dqpl-menu-trigger') ||
         !!closest(target, '.dqpl-menu-trigger');
+      const expandedDropdownItem = document.querySelector('[data-dropdown-expanded="true"]');
+      const menuExpanded = elements.menu.getAttribute('aria-expanded') === 'true';
 
-      if (isWithin || isHamburger || isLocked) {
-        return;
+      if (!isWithinDropdown && expandedDropdownItem) {
+        expandedDropdownItem.click();
       }
 
-      if (elements.menu.getAttribute('aria-expanded') === 'true') {
+      if (!isWithinSideBar && !isHamburger && !isLocked && menuExpanded) {
         onTriggerClick();
       }
-    });
+
+    }, true);
```

Comment:
this concerns me...seems like it'd be easy to not have to `useCapture` here

Reply:
Just double checked and I had to do this in order to get some tests working but that didn't turn out to be the issue. Will remove it.

Reply:
I was wrong here, `useCapture` ensures that this fires first which is what we want